### ISSUE TITLE
No proxy support

### ISSUE THREAD

**Issue by SadCodePoet** [adorsys/keycloak-config-cli#204]
**Describe the bug**
The KeycloakProvider does not heed the proxy settings. This disables tools like fiddler.

**To Reproduce**
1. Install fiddler and setup to capture/decode HTTPS traffic.
2. Start keycloak-config-cli with -DproxySet=true -DproxyHost=127.0.0.1 -DproxyPort=8888
Result: fiddler not able to capture traffic as Keycloak class creates RestEasy client without proxy support

**Expected behavior**
Traffic to Keycloak should be visible in Fiddler

**Environment (please complete the following information):**
 - Keycloak Version: [e.g. 9.0.0]
 - keycloak-config-cli Version: [1.4.0]
 - Java Version: [e.g. 8]

**Additional context**
The problem appears to be caused by the Keycloak implementation, which does not create the RestEasy client with proper proxy handling. A possible workaround is to use Reflection to force it like below:

```
public class KeycloakWithProxy {
    private static final Logger logger = LoggerFactory.getLogger(KeycloakImportProvider.class);
    public static Keycloak getInstance(String serverUrl, String realm, String username, String password, String clientId, String clientSecret, SSLContext sslContext, ResteasyJackson2Provider customJacksonProvider, boolean disableTrustManager, String authToken)  {
        Keycloak keycloak = null;
        try {
            Constructor<?>[] constructors = Keycloak.class.getDeclaredConstructors();
            for (Constructor<?> constructor : constructors) {
                logger.info(constructor.toGenericString());
                constructor.setAccessible(true);
                try {
                    keycloak = (Keycloak) constructor.newInstance(serverUrl, realm, username, password, clientId, clientSecret, "password", newRestEasyClient(customJacksonProvider, sslContext, disableTrustManager), authToken);
                } catch (Exception e)
                {
                    logger.error(e.getMessage());
                    continue;
                }
                break;
            }
        }
        catch (Exception e)
        {
            logger.error(e.getMessage());
        }
        return keycloak;
    }

    private static ResteasyClient newRestEasyClient(ResteasyJackson2Provider customJacksonProvider, SSLContext sslContext, boolean disableTrustManager) {
        ResteasyClientBuilder clientBuilder = (new ResteasyClientBuilder()).sslContext(sslContext).connectionPoolSize(10);
        if (disableTrustManager) {
            clientBuilder.disableTrustManager().hostnameVerification(ResteasyClientBuilder.HostnameVerificationPolicy.ANY);
        }

        if (customJacksonProvider != null) {
            clientBuilder.register(customJacksonProvider, 100);
        }

        clientBuilder.defaultProxy("127.0.0.1", 8888, "http");

        return clientBuilder.build();
    }
}
```

**Comment by jkroepke** [comment]
Did you try the java builtin proxy settings?

> https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html

TL;DR:

> -Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=8888

**Comment by jkroepke** [comment]
I'll will look into it. But before I do some nasty hacks like reflection, i will create an issue at upstream.

**Comment by jkroepke** [comment]
Ref: https://issues.redhat.com/browse/KEYCLOAK-15775

**Comment by jkroepke** [comment]
Meanwhile take a look at #206 if its fit for you. You could find prebuilt artifacts in the Actions section.
